tagstudio/src/qt/widgets/audio_player.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (C) 2024 Travis Abendshien (CyanVoxel).
+# Licensed under the GPL-3.0 License.
+# Created for TagStudio: https://github.com/CyanVoxel/TagStudio
+
+import logging
+import typing
+from pathlib import Path
+
+from PySide6.QtCore import (
+    Qt,
+    QUrl,
+)
+from PySide6.QtGui import QIcon, QPixmap
+from PySide6.QtMultimedia import QAudioOutput, QMediaDevices, QMediaPlayer
+from PySide6.QtWidgets import QHBoxLayout, QPushButton, QSlider, QVBoxLayout, QWidget
+
+if typing.TYPE_CHECKING:
+    from src.qt.ts_qt import QtDriver
+
+
+class AudioPlayer(QWidget):
+    """A basic audio player widget.
+
+    This widget is enabled if the selected
+    file type is found in the AUDIO_TYPES list.
+    """
+
+    def __init__(self, driver: "QtDriver") -> None:
+        super().__init__()
+        self.driver = driver
+        self.filepath: Path | None = None
+        self.base_size: tuple[int, int] = (266, 75)
+
+        self.setMinimumSize(*self.base_size)
+        self.setMaximumSize(*self.base_size)
```

Comment:
I agree setting the size explicitly isn't the best approach (didn't know it would have those side effects). I think I'm probably going to have to look into a different way to size the widget though. 

When I remove the min/max size, the audio widget grows/shrinks as expected. However, after a certain point the UI looks overstretched. I'll do some more research soon.

Also open to suggestions if you think of any.

Reply:
can you help me to reproduce that issue?

Reply:
Sure. If you apply your change and resize the audio player widget, the buttons/slider will space out (note: I've made a few UI tweaks).

This is normal:
![Screenshot_20241115_160547](https://github.com/user-attachments/assets/ca55c64c-6d93-4426-8c4a-4fef46793190)

This is after resizing the audio player widget:
![Screenshot_20241115_160338](https://github.com/user-attachments/assets/2b597e5d-5d9b-449f-b6b2-369589a02ceb)

I think it would be nice if the buttons stay more compact. Thoughts?

Reply:
What if I just do `setFixedHeight()` on the audio player?

Not sure how I feel about it, but I don't think it would affect the other widgets in the preview panel.

Reply:
that is the expected behavior. the `self.base_layout` (`QVBoxLayout`) stretches the widgets as the `AudioPlayer` widget resizes.
and yes, using `setFixedHeight()` will fix that. i didn’t notice any issues with other widgets in the preview panel when setting a fixed height.